### PR TITLE
Docs dataframe joins

### PR DESCRIPTION
**Pull request by mrocklin**
- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

File: docs/source/dataframe-joins.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Dask will have to read and write data to disk, which may cause other
+performance costs.
+
+These problems are solvable, but will be significantly slower than many other
+operations.  They are best avoided if possible.
+
+Large to Small Joins
+--------------------
+
+Many join or merge computations combine a large tqble with one small one.  If
```
Comment (TomAugspurger) [inline docs/source/dataframe-joins.rst]: typo: "tqble"

File: docs/source/dataframe-joins.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+normal Pandas DataFrame then the computation can proceed in an embarrassingly
+parallel way, where each partition of the large dataframe is joined against the
+single small table.  This incurs almost no overhead relative to Pandas joins.
+
+If your smaller table can easily fit in memory, then you might want to ensure
+that it is a single partition with the following
+
+.. code-block:: python
+
+    small.repartition(npartitions=1)
```
Comment (TomAugspurger) [inline docs/source/dataframe-joins.rst]: `small = small.repartition(npartitions=1)`?

File: docs/source/dataframe-joins.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    result = big.merge(small)
+
+Sorted Joins
+------------
+
+The Pandas merge API supports the ``left_index=`` and ``right_index=`` options
+to perform joins on the index.  For Dask DataFrames these keyword options hold
+special significance if the index has known divisions (see :doc:`dataframe
```
Comment (TomAugspurger) [inline docs/source/dataframe-joins.rst]: If you add a label above Partitions in `dataframe-design` like

```
.. _dataframe-design-partitions

Partitons
---------
```

you'll be able to reference like \`:ref:\`dataframe-design-partitions` I think, to link directly to the relevant section.

File: docs/source/dataframe-joins.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+design <dataframe-design>`).  In this case the dataframe partitions are
```
Comment (jrbourbeau) [inline docs/source/dataframe-joins.rst]: Nitpick for consistency 
```suggestion
design <dataframe-design>`).  In this case the DataFrame partitions are
```

File: docs/source/dataframe-joins.rst [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+These problems are solvable, but will be significantly slower than many other
+operations.  They are best avoided if possible.
+
+Large to Small Joins
+--------------------
+
+Many join or merge computations combine a large tqble with one small one.  If
+the small table is either a single partition Dask DataFrame or even just a
+normal Pandas DataFrame then the computation can proceed in an embarrassingly
+parallel way, where each partition of the large dataframe is joined against the
```
Comment (jrbourbeau) [inline docs/source/dataframe-joins.rst]: Same as other comment 
```suggestion
parallel way, where each partition of the large DataFrame is joined against the
```

File: docs/source/dataframe-joins.rst [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    result = big.merge(small)
+
+Sorted Joins
+------------
+
+The Pandas merge API supports the ``left_index=`` and ``right_index=`` options
+to perform joins on the index.  For Dask DataFrames these keyword options hold
+special significance if the index has known divisions
+(see :ref:`dataframe-design-partitions`).
+In this case the dataframe partitions are aligned along these divisions (which
```
Comment (jrbourbeau) [inline docs/source/dataframe-joins.rst]:
```suggestion
In this case the DataFrame partitions are aligned along these divisions (which
```

**Comment by jrbourbeau** [comment]
Thanks!

**Pull request merged**